### PR TITLE
Fix/8882

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This brings a fix for editting many to many relationships, as well as adding a lot more checking around what can and can't be a junction table. For Budibase to be opinionated about generating many to many relationships, the table needs to be a well defined junction table - this is discussed in the issue.

The majority of this issue was updating table definitions to include the presence constraint, which means that Budibase is now able to pre-populate the `required` attribute of a column based on a `NOT NULL` option in the SQL DDL of a table schema.

I've also added the configuration used to create this bad table scenario to the integration scripts in the server, for future use in a test case for this.

Addresses: 
- #8882 

## Screenshots
How the error appears for the table configuration described in the issue.
image

### CODE REVIEW

File: packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte [hunk 1]
```diff
@@ -59,6 +76,10 @@
     if ($touched.through && isMany && !fromRelate.through) {
       errObj.through = tableNotSet
     }
+    if ($touched.through && invalidThroughTable(fromRelate)) {
+      errObj.through =
+        "Table contains non-nullable columns which aren't generated"
```
Comment (shogunpurple) [inline packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte:81]: I would potentially flip this to make it easier to understand:

`Table PK must have auto generated` or similar.

**Review by shogunpurple — APPROVED** [review]
LGTM - just think the error message could be a little clearer. Good job

File: packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte (hunk 1 above)
Comment (mike12345567) [reply packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte:81]: In this case its specifically about non-nullable fields, its not to do with the PK. For example you could have a whole bunch of non-nullable fields in your junction table, Budibase will never be able to fill them in - creation of rows will always fail.

Comment (shogunpurple) [reply packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte:81]: Gotcha, in that case maybe:
`Ensure all columns in table are nullable or auto generated`

**Pull request merged**